frontend/src/components/Admin/AdminShoutouts/AdminShoutouts.tsx

```diff
@@ -1,41 +1,92 @@
 import React, { useState, useEffect } from 'react';
-import { Item, Card } from 'semantic-ui-react';
+import { Form, Item, Card } from 'semantic-ui-react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { Emitters } from '../../../utils';
 import { ShoutoutsAPI, Shoutout } from '../../../API/ShoutoutsAPI';
 import styles from './AdminShoutouts.module.css';
 
 const AdminShoutouts: React.FC = () => {
   const [shoutouts, setShoutouts] = useState<Shoutout[]>([]);
+  const [displayShoutouts, setDisplayShoutouts] = useState<Shoutout[]>([]);
   useEffect(() => {
-    ShoutoutsAPI.getAllShoutouts().then((shoutouts) => setShoutouts(shoutouts));
+    ShoutoutsAPI.getAllShoutouts().then((shoutouts) => {
+      setShoutouts(shoutouts);
+      setDisplayShoutouts(shoutouts);
+    });
   }, []);
+  const [earlyDate, setEarlyDate] = useState<Date>(new Date('2022/01/01'));
+  const [lastDate, setLastDate] = useState<Date>(new Date());
 
   const fromString = (shoutout: Shoutout): string => {
+    const dateString = ` (Date: ${new Date(shoutout.timestamp).toDateString()})`;
     if (!shoutout.isAnon) {
       const { giver } = shoutout;
-      return `From: ${giver?.firstName} ${giver?.lastName} (${giver.email})`;
+      return `From: ${giver?.firstName} ${giver?.lastName} ${dateString}`;
+    }
+    return `From: Anonymous${dateString}`;
+  };
+
+  const updateShoutouts = () => {
+    if (lastDate < earlyDate) {
+      Emitters.generalError.emit({
+        headerMsg: 'Invalid Date Range',
+        contentMsg: 'Please make sure the latest shoutout date is after the earliest shoutout date.'
+      });
+    } else {
+      setDisplayShoutouts(
+        shoutouts.filter(
+          (shoutout) =>
+            new Date(shoutout.timestamp) >= earlyDate && new Date(shoutout.timestamp) <= lastDate
+        )
+      );
     }
-    return 'From: Anonymous';
   };
 
   return (
-    <div className={styles.shoutoutsContainer}>
-      {shoutouts.length === 0 ? (
-        <Card className={styles.noShoutoutsContainer}>
-          <Card.Content>No shoutouts.</Card.Content>
-        </Card>
-      ) : (
-        <Item.Group divided>
-          {shoutouts.map((shoutout, i) => (
-            <Item key={i}>
-              <Item.Content>
-                <Item.Header>{`To: ${shoutout.receiver}`}</Item.Header>
-                <Item.Meta>{fromString(shoutout)}</Item.Meta>
-                <Item.Description>{shoutout.message}</Item.Description>
-              </Item.Content>
-            </Item>
-          ))}
-        </Item.Group>
-      )}
+    <div>
+      <Form className={styles.shoutoutForm}>
+        <h2 className={styles.formTitle}>Select date range to display shoutouts:</h2>
+        <Form.Group width="equals">
+          <DatePicker
```

Comment:
Might be good to make a component that contains what these two DatePickers have in common, and just takes in the selected date and the setter function. Just to consolidate the two DatePicker's since they're really similar.

Reply:
@andrew032011 Makes sense. I believe I addressed most of your comments, so please review if you get a chance!

Reply:
Looks great! Thoughts on moving the date to be inline with the `To: ...` text? So like, top right corner.
 
<img width="1056" alt="Screen Shot 2022-11-04 at 1 17 19 PM" src="https://user-images.githubusercontent.com/59291082/200036820-22078893-881f-4dad-b5cb-b76e137e3e0c.png">

Reply:
I agree. I was having issues with formatting before, but I fixed it now!